examples/EmailerAgent/emailer/agent.py

```diff
@@ -0,0 +1,419 @@
+        try:
+            server = smtplib.SMTP(self.current_config.get('smtp_address', None),self.current_config.get('smtp_port', None))
+            server.ehlo()
+            server.starttls()
+            #stmplib docs recommend calling ehlo() before & after starttls()
+            server.ehlo()
+            server.login(self.current_config.get('smtp_username', None), self.current_config.get('smtp_password', None))
+            #server.sendmail(SENDER, RECIPIENT, msg.as_string())
+            server.close()
+        except  Exception as e:
+            print ("Error: ", e)
+        else:
+            print ("Email sent!")
+
+    def configure_main(self, config_name, action, contents):
+        """
+        The main configuration callback.
+
+        :param config_name:
+        :param action:
+        :param contents:
+        """
+        self.current_config = self.default_config.copy()
+        self.current_config.update(contents)
+
+        self.current_config['allow_frequency_seconds'] = self.current_config.get(
+            'allow_frequency_minutes', 60) * 60
+
+        smtp_address = self.current_config.get('smtp_address', None)
+        smtp_port = self.current_config.get('smtp_port', None)
+        smtp_username = self.current_config.get('smtp_username', None)
+        smtp_password = self.current_config.get('smtp_password', None)
+
+        if action == "NEW":
+            try:
+                with gevent.with_timeout(3, self._test_smtp_address, smtp_address,smtp_port,smtp_username,smtp_password):
+                    pass
+            except Exception as e:
+                self.vip.health.set_status(STATUS_BAD, "Invalid SMTP Address")
+
+    def on_email_message(self, peer, sender, bus, topic, headers, message):
+        """
+        Callback used for sending email messages through the pubsub bus.
+
+        Either the from_address and to_addresses can be ommitted if they
+        are specified in the configuration store/file.  If they are to be used
+        the following block shows the format for usage.
+
+        .. code-block:: json
+
+            {
+                "from-address": 'foo@bar.com',
+                "to-addresses": ['alpha.beta@fo.com', 'bob-and-joe@bim.com']
+            }
+
+        ** In the above code to-addresses can be a singe email address as well**
+
+        The message must be a dictionary containing a subject and a message.
+
+        .. code-block:: json
+
+            {
+                "subject": "I am a happy camper",
+                "message": "This is a big long string message that I am sending"
+            }
+
+        :param peer:
+        :param sender:
+        :param bus:
+        :param topic:
+        :param headers:
+        :param message:
+        """
+        from_address = self.current_config.get('from_address')
+        to_addresses = self.current_config.get('to_addresses')
+        smtp_port = self.current_config.get('smtp_port')
```

Comment:
These should already be present in member variables (except for possibly the from/to addresses)